src/test/scala/strawman/collection/test/UnfoldTest.scala

```diff
@@ -0,0 +1,98 @@
+package strawman
+package collection
+package test
+
+import strawman.collection.mutable.Builder
+
+import scala.{Int, None, Option, Some, Unit}
+import scala.Predef.implicitly
+import java.lang.String
+
+import org.junit.Test
+
+/**
+  * Builds a collection of type `C`.
+  *
+  * @tparam C Type of the collection to produce (e.g. `List[Int]`, `TreeMap[Int, String]`, etc.)
+  */
+trait Unfolder[C] {
+
+  val bt: BuildTo[C]
```

Comment:
These tests check that it is possible to implement generic “unfold” operations without touching the core of the collections.

We could use the same approach as in the `BuildFrom.factory` methods but that requires to write 4 methods, corresponding to the 4 variants of collection factories ((ordered vs unordered) × (unary vs binary type constructor)). Here I chose to explore a way to define one single `unfold` method that could be used to produce lists, maps and ordered collections as well. I’m relying on an auxiliary type named `BuildTo`, which plays a similar role as `BuildFrom`: provide builders in a type-directed way. In the case of `BuildTo` the goal is to get a `Builder[Int, List[Int]]` from a `BuildTo[List[Int]]`, or a `Builder[(Int, String), TreeMap[Int, String]]` from a `BuildTo[TreeMap[Int, String]]`.